cmd/swapd/main.go

```diff
@@ -409,6 +409,7 @@ func (d *daemon) make(c *cli.Context) error { //nolint:gocyclo
 	if err != nil {
 		return err
 	}
+	defer ec.Close()
```

Comment:
remove this? it's passed into `newBackend` which is passed into `extethclient.NewEthClient`

Reply:
The backend goes out of scope at the end of this function too. It is not part of the `daemon` type's fields.  The extended ethclient doesn't close the connection, because it didn't create the connection. The XMR client closer is also handled as a defer in this, somewhat oversized, `make` method.  I'm open to other ideas, but I think this is the logical place to close it at the current time.

Reply:
the backend goes out of scope, but it's passed into `getProtocolInstances` which uses the backend for both maker/taker. am I missing something? it seems like the ethclient is still used at that point

Reply:
I agree, it is usually best to destroy from the outside to the inside (reverse order of creation). This function is already destroying the XMR client, which is analogous to the eth client, so it is not doing that already. I have some ideas on how to rewrite this code that I can put into a future PR. I removed the `ec.Close()` for now.